### ISSUE TITLE
I get the balance of the last item in the waiting list from the API, but it is inconsistent with the waiting amount in my backend. What is the problem?

### ISSUE THREAD

**Issue by a785817258** [allegro/allegro-api#9490]
![image](https://github.com/user-attachments/assets/14f64994-cf2c-4c29-a38a-c9d2679c8878)

**Comment by AureliuszBrussy** [comment]
The page shows the current combined balance for the two payment operators (PayU and Przelewy24). In response from the public API, we return separate balances for both payment operators and the balance as of the moment the payment operation occurred. It is difficult for me to address any differences (which, of course, should not occur). Please report it via the [contact form](https://allegro.pl/pomoc/kontakt) after logging in to the related account.

**Comment by a785817258** [comment]
Is it related to the time zone? My local time is Beijing time.

**Comment by AureliuszBrussy** [comment]
No, I don't think it's related to the time zone. In the API, we return all data in universal time (zulu).

**Comment by a785817258** [comment]
![image](https://github.com/user-attachments/assets/cd3c4204-f471-4fba-ba09-45f2a1ad723a)
![image](https://github.com/user-attachments/assets/a1c91123-c8fc-4f6e-bfd9-f3127613c5b7)
Or is there something wrong with the way I request it?

**Comment by a785817258** [comment]
Also, why is there no sorting field for this list? What attribute is used to sort the data each time it is retrieved?

**Comment by AureliuszBrussy** [comment]
We sort by default from the most recent event. 

You need to pay attention to the fact that there are three payment operators (PayU, Przelewy24, Allegro Finanse) that potentially process the payment. Therefore, there are 3 separate balances. You can filter the results for a given operator using the query parameter `wallet.paymentOperator`.

**Comment by a785817258** [comment]
------------------------------------AVAILABLE--------------------------------------------------------
Allegro Api：/payments/payment-operations
Allegro request params：{"offset":0,"limit":1,"wallet.type":"AVAILABLE"}
Allegro response：{"paymentOperations":[{"type":"DEDUCTION_CHARGE","group":"OUTCOME","wallet":{"paymentOperator":"PAYU","type":"AVAILABLE","balance":{"amount":"947.19","currency":"PLN"}},"occurredAt":"2024-07-24T10:30:26.104Z","value":{"amount":"-9.34","currency":"PLN"},"marketplaceId":"allegro-pl","deduction":{"id":"abe9d76b-6a57-4855-8bfc-b0b6852f0b2c"}}],"count":1,"totalCount":8405}

------------------------------------WAITING -------------------------------------------------------------
Allegro Api：/payments/payment-operations
Allegro request params：{"offset":0,"limit":1,"wallet.type":"WAITING"}
Allegro response：{"paymentOperations":[{"type":"CONTRIBUTION","group":"INCOME","wallet":{"paymentOperator":"PAYU","type":"WAITING","balance":{"amount":"6831.70","currency":"PLN"}},"occurredAt":"2024-06-14T07:23:51.961Z","value":{"amount":"97.99","currency":"PLN"},"marketplaceId":"allegro-pl","payment":{"id":"eeb4b42a-2a1e-11ef-afcb-e3671250ed78"},"participant":{"id":"671380","companyName":null,"login":"aniakania","firstName":"Anna","lastName":"Kania","address":{"street":"Spokojna 15","city":"Zielonki Wieś","postCode":"05-082"}}}],"count":1,"totalCount":305}
![image](https://github.com/user-attachments/assets/07927e6f-6573-4571-bab9-121a96f62a0f)
I see that the first item in the document returns the latest data, why doesn't WAITING return the latest data?

**Comment by AureliuszBrussy** [comment]
@a785817258 it should respond with the latest event. Can you send the trace-id of this request? I'll check it out. You can get trace-id in response header.

**Comment by a785817258** [comment]
@AureliuszBrussy f605da0b4adb648c

**Comment by AureliuszBrussy** [comment]
Thank you - we will check it out. I will get back to you.

**Comment by AureliuszBrussy** [comment]
@a785817258 this is inconsistency on our side. We are preparing a fix so that we will return the entries according to the expected order (i.e. according to the occuredAt date). I will let you know when the fix is implemented.
